src/components/src/handoff-message/index.js

```diff
@@ -30,9 +30,6 @@ export default function HandoffMessage() {
 				setHandoffMessage( false );
 			}
 		}, 100 );
-
-		// Clean up the notification when unmounting.
-		return () => window.localStorage.removeItem( HANDOFF_KEY );
```

Comment:
This was removed because it instantly removed the handoff message when redirecting from a different tab in the same wizard. That happens because it unmounts the component from the origin tab, given it's a rerender on the same react app.